### PR TITLE
Upgrade Envoy to v1.23.0 and misc changes

### PR DESCRIPTION
**Pull request by mbana**
Upgrade Envoy to `docker.io/envoyproxy/envoy:v1.23.0`.

Remove `resources` (limits) from `development/testing/manifests/2_envoyfleet.yaml`.

Signed-off-by: Mohamed Bana <mohamed@bana.io>

---

This PR...

## Changes

-

## Fixes

-

## Checklist

- [ ] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

**Review by jasmingacic — APPROVED** [review]
Looks legit.

**Comment by aabedraba** [comment]
😍

File: config/samples/gateway_v1_envoyfleet.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ kind: EnvoyFleet
 metadata:
   name: default
 spec:
-  image: "envoyproxy/envoy-alpine:v1.20.0"
+  image: "docker.io/envoyproxy/envoy:v1.23.0"
```
Comment (jasmingacic) [inline config/samples/gateway_v1_envoyfleet.yaml:6]: Comparing these two images I see that alpine image is at 28MB where plain envoy is 50MB. But the only drawback of the alpine images is that it doesn't seem to be supported anymore.

**Review by jasmingacic — CHANGES_REQUESTED** [review]
I see that CI passes but what I would like to see is that we split this PR into two smaller PRs.
One to handle container image update and the other one to include changes like Validate -> ValidateAll and similar. 

Just to keep them separate in case we need to revert one of these changes.

What do you think?

@kylehodgetts what do you think?

**Comment by jasmingacic** [comment]
I stand corrected it looks like that CI doesn't agree with me

File: config/samples/gateway_v1_envoyfleet.yaml (hunk 1 above)
Comment (mbana) [reply config/samples/gateway_v1_envoyfleet.yaml:6]: That's correct. The Alpine image does not receive updates any more.

**Comment by mbana** [comment]
> I see that CI passes but what I would like to see is that we split this PR into two smaller PRs. One to handle container image update and the other one to include changes like Validate -> ValidateAll and similar.
> 
> Just to keep them separate in case we need to revert one of these changes.
> 
> What do you think?
> 
> @kylehodgetts what do you think?

I'll make those changes.

**Review by jasmingacic — APPROVED** [review]
CI is good with it.

I'd say let's ship it.

**Pull request merged**